Refactors medical stacks used on simple animals.

harmonyn opened the pull request:
Simple animals had an entirely separate and shitty piece of code that handled healing with medical stacks, and it overrode the medical.dm code. This fucked up everything as well as causing some mobs that shouldn't be able to be healed, to be able to. 

This PR adds a new var for simple animals, healable, that determines whether medical stacks work on them. It defaults to 1.

Current list of simple_animals that cannot be healed with medical stacks:

Drones. Fixes https://github.com/tgstation/-tg-station/issues/9760.
Constructs.
Shades.
Slimes.
Revenants.
Slaughter demons.
Morphs.
Hivebots.
Statues. (AS IF THEY NEED IT, FUCK)

MERGE CONFLICT FUCKING FIXED, FUCK

Commit:
You can no longer use bruise packs, ointment or other medical stack items on drones.

RemieRichards commented:
this code is illogical.
M is a carbon. drones are not carbon. the ```if(!istype(M))``` should be working.

bgobandit commented:
that part of the code wasn't mine. for whatever reason it seems it simply doesn't work -- simple animals aren't carbons, but you can use bruise packs/etc on them

my suspicion is it's supposed to be a sanity test

RemieRichards commented:
1. Yes I'm aware
2. Yes it is a sanity check, checking the mob is of the right type, but it's failed.

Please try expanding the ```if(!istype(M))``` to ```if(!istype(M, /mob/living/carbon))``` and see if that works.

Incoming5643 commented:
There's handling later in the code for healing simple animals, which is intended. This check, if it ever worked, was obsoleted and apparently forgotten.

bgobandit commented:
is that code somewhere other than medical.dm

Incoming5643 commented:
Yep, there's cruddy, non OOP code in https://github.com/tgstation/-tg-station/blob/909fcdc9a532f452817f8da85d4b67f46fe156dd/code/modules/mob/living/simple_animal/simple_animal.dm that handles healing for simple animals

bgobandit commented:
WELL THAT'S WHY THIS IS FUCKED UP THEN, WHY WOULD SOMEONE EVEN DO THAT

brb refactoring this to not suck

Commit:
Refactors healing simple animals with bruise packs.

Commit:
g merge conflicts. remote-tracking branch 'upstream/master' into dronepax

Conflicts:
	code/modules/mob/living/simple_animal/revenant/revenant.dm
	code/modules/mob/living/simple_animal/slaughter/slaughter.dm

Cheridan commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -25,8 +25,8 @@
-	if(!istype(M))
-		user << "<span class='danger'>You don't know how to apply \the [src] to [M]...</span>"
+	if(!istype(M, /mob/living/carbon) && !istype(M, /mob/living/simple_animal))
+		user << "<span class='danger'>You don't know how to apply \the [src] to [M]</span>..."
```
Make this a period. We use ellipses to indicate an action that takes time to do.

Miauw62 replied:
also, make sure your spans wrap around the entire message

Commit:
fixes ellipsis

bgobandit commented:
k
